pkg/crc/machine/machine.go

```diff
@@ -757,7 +757,7 @@ func updateSSHKeyPair(sshRunner *crcssh.Runner) error {
 	if err != nil {
 		return err
 	}
-	cmd := fmt.Sprintf("echo '%s' > /home/core/.ssh/authorized_keys", publicKey)
+	cmd := fmt.Sprintf("echo '%s' > /home/core/.ssh/authorized_keys; chmod 644 /home/core/.ssh/authorized_keys", publicKey)
```

Comment:
> but in case of fedora coreos the file's initial permissions are too open

This would be good to report a bug against fedora coreos so that they can fix it. Or maybe it's just that on fedora coreos, ~/.ssh/authorized_keys initially does not exist, and when a new file is created, it's getting 664 permissions by default?

Reply:
Need to check again, i did check for this but forgot.. :( if its preexisting with wrong permissions will report a bug

Reply:
> Or maybe it's just that on fedora coreos, ~/.ssh/authorized_keys initially does not exist, and when a new file is created, it's getting 664 permissions by default?

This is the case, and i think since the suggested way is to inject keys using ignition which uses the `authorized_keys.d` directory not sure if this should be reported.

Reply:
I don't think it needs to be reported, creating a new file will honour your umask settings, and that's about it. If the default permissions don't match what you need, then you get to change them yourself, which is what this commit is doing